src/lib/shadow-shim-helper-rs/src/rootedcell/rc.rs

```diff
@@ -202,15 +204,32 @@ impl<T> RootedRc<T> {
         }
     }
 
-    /// Safely drop this object, dropping the internal value if no other
-    /// references to it remain.
-    ///
-    /// Instances that are dropped *without* calling this method cannot be
-    /// safely cleaned up. In debug builds this will result in a `panic`.
-    /// Otherwise the underlying reference count will simply not be decremented,
-    /// ultimately resulting in the enclosed value never being dropped.
+    /// Drop the `RootedRc`, and return the inner value if this was the last
+    /// strong reference.
     #[inline]
-    pub fn safely_drop(self, root: &Root) {
+    pub fn into_inner(this: Self, root: &Root) -> Option<T> {
+        this.common.safely_drop(root, RefType::Strong)
+    }
```

Comment:
Do we have anything in a `RootedRc` in Shadow that relies on the fact that it doesn't (well didn't previously) move? For example I think I remember the host and host shmem lock having circular references with unsafe, relying on the fact that the lock would never move? I don't have any specific issue in mind, but just wanted to leave a comment to maybe double check that this change doesn't break any assumption of that code.